src/metrics.rs

```diff
@@ -2440,9 +2506,18 @@ impl GooseAttack {
             }
             if !self.configuration.report_file.is_empty() {
                 if let Some(starting) = self.metrics.starting {
-                    merge_request.record_requests_per_second(
-                        (Utc::now().timestamp() - starting.timestamp()) as usize,
+                    let second_since_start =
+                        (Utc::now().timestamp() - starting.timestamp()) as usize;
+
+                    merge_request.record_requests_per_second(second_since_start);
+                    merge_request.record_average_response_time_per_second(
+                        second_since_start,
+                        request_metric.response_time,
                     );
+
+                    if !request_metric.success {
```

Comment:
> Is there ever a time success is false but the error text is empty?

I have no idea :) I am not  familiar enough with Goose to be able to answer that. 

With that said - I do think that using a bool instead of a string in a condition makes more sense.

Reply:
Reviewing the code in question, I see it's done that way as we store the actual text of the error for a final report: but that's only in the latter part of that function, so likely we should start by checking the boolean as you've done. In other words: what you're doing is correct, thanks.

Reply:
👍 I updated that condition too.